drivers/hmis/spec/jobs/auto_exit_job_spec.rb

```diff
@@ -0,0 +1,103 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'rails_helper'
+
+RSpec.describe Hmis::AutoExitJob, type: :model do
+  # Probably other specs aren't cleaning up:
+  before(:all) { cleanup_test_environment }
+
+  let!(:ds1) { create(:hmis_data_source) }
+  let!(:u1) { create :hmis_hud_user, data_source: ds1, user_email: 'test@example.com' }
+  let!(:o1) { create :hmis_hud_organization, data_source: ds1, user: u1 }
+
+  describe 'with residential projects' do
+    let!(:p1) { create :hmis_hud_project, data_source: ds1, organization: o1, user: u1, project_type: 0 }
+    let!(:c1) { create :hmis_hud_client, data_source: ds1, user: u1 }
+    let!(:aec) { create :hmis_auto_exit_config, length_of_absence_days: 30 }
+
+    it 'should exit correctly for a bed night' do
+      e1 = create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c1, user: u1, entry_date: Date.today - 2.months
+      e2 = create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c1, user: u1, entry_date: Date.today - 2.months
+      create :hmis_hud_service, data_source: ds1, client: c1, enrollment: e1, user: u1, record_type: 200
+      s2 = create :hmis_hud_service, data_source: ds1, client: c1, enrollment: e2, user: u1, record_type: 200, date_provided: Date.today - 31.days
+
+      Hmis::AutoExitJob.perform_now
+
+      expect(Hmis::Hud::Enrollment.exited).to include(e2)
+      expect(Hmis::Hud::Enrollment.exited).not_to include(e1)
+      expect(e2.exit).to have_attributes(auto_exited: be_present, exit_date: s2.date_provided + 1.day, destination: 30)
+      expect(e2.custom_assessments).to contain_exactly(have_attributes(assessment_date: s2.date_provided + 1.day, data_collection_stage: 3))
```

Comment:
Maybe more readable:
`expect(e2.exit_assessment&.assessment_date).to eq(s2.date_provided + 1.day)`